x-pack/plugins/serverless_search/server/routes/api_key_routes.ts

```diff
@@ -7,27 +7,40 @@
 
 import { schema } from '@kbn/config-schema';
 import { RouteDependencies } from '../plugin';
+import { errorHandler } from '../utils/error_handler';
 
 export const registerApiKeyRoutes = ({ logger, router, getSecurity }: RouteDependencies) => {
   router.get(
     {
       path: '/internal/serverless_search/api_keys',
       validate: {},
     },
-    async (context, request, response) => {
+    errorHandler(logger)(async (context, request, response) => {
       const core = await context.core;
       const { client } = core.elasticsearch;
       const user = core.security.authc.getCurrentUser();
       if (user) {
-        const apiKeys = await client.asCurrentUser.security.getApiKey({ username: user.username });
-        const validKeys = apiKeys.api_keys.filter(({ invalidated }) => !invalidated);
-        return response.ok({ body: { apiKeys: validKeys } });
+        const privileges = await client.asCurrentUser.security.hasPrivileges({
+          cluster: ['manage_own_api_key'],
+        });
+        const canManageOwnApiKey = privileges?.cluster.manage_own_api_key;
+
+        try {
+          const apiKeys = await client.asCurrentUser.security.getApiKey({
+            username: user.username,
+          });
+
+          const validKeys = apiKeys.api_keys.filter(({ invalidated }) => !invalidated);
+          return response.ok({ body: { apiKeys: validKeys, canManageOwnApiKey } });
+        } catch {
+          return response.ok({ body: { apiKeys: [], canManageOwnApiKey } });
```

Comment:
It feels like we are silencing a possible error on `getApiKey` part.

Reply:
Yup, but that's on purpose. ManageOwnApiKey doesn't get you the privilege to list API keys, so we'd have to check that separately. This swallows some errors, but those should generally be transient anyway.